PyInstaller/hooks/hook-shapely.py

```diff
@@ -23,12 +24,22 @@
 
 binaries = []
 if compat.is_win:
+    # Search conda directory if conda is active, then search standard
+    # directory. This is the same order of precidence used in shapely.
+    lib_paths = ''
     if compat.is_conda:
-        lib_dir = os.path.join(compat.base_prefix, 'Library', 'bin')
-    else:
-        lib_dir = os.path.join(pkg_dir, 'DLLs')
-    dll_files = ['geos_c.dll', 'geos.dll']
-    binaries += [(os.path.join(lib_dir, f), '.') for f in dll_files]
+        lib_paths = os.path.join(compat.base_prefix, 'Library', 'bin')
+        lib_paths += os.pathsep
+    lib_paths += os.path.join(pkg_dir, 'DLLs')
+    original_path = os.environ['PATH']
+    os.environ['PATH'] = lib_paths + os.pathsep + original_path
+    dll_path = find_library('geos_c')
```

Comment:
Thsi should be wrapped into `try … finally` and restore the original PATH afterwards. Otherwise this hook will haveside-effects which will be extremly hard to debug.

Reply:
Done